### DISCUSSION TITLE
ALERT: ALERT_TS_NARROWED!

### DISCUSSION THREAD

**Discussion by rootBiao** [strongswan/strongswan#1835] [Q&A]
![image](https://github.com/strongswan/strongswan/assets/87700793/a87e556b-e86d-41a8-a8d1-450460498d9e)
Clinet received this warning, but I don't know what it means.

**Comment by strongX509** [comment]
**`ALERT_TS_NARROWED`** is a signal used by strongSwan on its internal bus to announce that the received traffic selector has been narrowed down which is a regular IKEv2 operation. So this is normal behaviour, not an error.

The **`ALERT: ALERT_TS_NARROWED`** debug output line has not been generated by a standard strongSwan plugin but rather is being issued either directly by a print statement added to the strongSwan source code or by a proprietary libcharon plugin listening on the internal signalling bus.

**Reply by rootBiao** [reply]
thanks!

**Comment by rootBiao** [comment]
这是来自QQ邮箱的假期自动回复邮件。你的邮件，陈陆彪已经收到。
